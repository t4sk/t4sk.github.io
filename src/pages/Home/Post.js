import React from "react"
import PropTypes from "prop-types"
import { Image } from "semantic-ui-react"
import styles from "./Post.module.css"

export function Post(props) {
  const { youTubeUrl, thumbnailUrl, title, createdAt } = props

  return (
    <a className={styles.component} href={youTubeUrl} target="__blank">
      <div className={styles.image}>
        <Image src={thumbnailUrl} fluid />
      </div>
      <div className={styles.content}>
        <div className={styles.title}>{title}</div>
        <div className={styles.date}>{createdAt}</div>
      </div>
    </a>
  )
}

Post.propTypes = {
  youTubeUrl: PropTypes.string.isRequired,
  thumbnailUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
}

export default Post
