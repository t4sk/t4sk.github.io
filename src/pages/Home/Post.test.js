import React from "react"
import { shallow } from "enzyme"
import Post from "./Post"

const props = {
  thumbnailUrl:
    "https://i.ytimg.com/vi/hMwdd664_iw/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLDZXbU3wGs1vpsnWX06BEHw9TookQ",
  youTubeUrl: "https://www.youtube.com/watch?v=hMwdd664_iw",
  title: "Learn Solidity - Hello World",
  createdAt: "2019-08-20",
}

test("it renders", () => {
  const component = shallow(<Post {...props} />)

  expect(component).toMatchSnapshot()
})
