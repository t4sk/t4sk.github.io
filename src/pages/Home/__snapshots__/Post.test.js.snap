// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<a
  className="component"
  href="https://www.youtube.com/watch?v=hMwdd664_iw"
  target="__blank"
>
  <div
    className="image"
  >
    <Image
      as="img"
      fluid={true}
      src="https://i.ytimg.com/vi/hMwdd664_iw/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLDZXbU3wGs1vpsnWX06BEHw9TookQ"
      ui={true}
    />
  </div>
  <div
    className="content"
  >
    <div
      className="title"
    >
      Learn Solidity - Hello World
    </div>
    <div
      className="date"
    >
      2019-08-20
    </div>
  </div>
</a>
`;
